File: JPAArtConnect/src/main/java/com/skilldistillery/artisphere/entities/Artwork.java
package com.skilldistillery.artisphere.entities;

import java.time.LocalDateTime;
import java.util.Objects;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity
public class Artwork {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String title;

    private String artist;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;


	@ManyToOne
    @JoinColumn(name = "museum_id")
    private Museum museum;

    @Column(name = "creation_year")
    private int creationYear;

    private String description;

    private String image;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    
    public User getUser() {
  		return user;
  	}

  	public void setUser(User user) {
  		this.user = user;
  	}
    
    
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getArtist() {
        return artist;
    }

    public void setArtist(String artist) {
        this.artist = artist;
    }



    public Museum getMuseum() {
        return museum;
    }

    public void setMuseum(Museum museum) {
        this.museum = museum;
    }

    public int getCreationYear() {
        return creationYear;
    }

    public void setCreationYear(int creationYear) {
        this.creationYear = creationYear;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }

  
    @Override
    public String toString() {
        return "Artwork [id=" + id + ", title=" + title + ", artist=" + artist + ", creationYear=" + creationYear + "]";
    }

  
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Artwork artwork = (Artwork) o;
        return id == artwork.id;
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
    
    
   
    
}

File: NgArtWorld/src/app/models/artwork.ts
import { Museum } from "./museum";
import { User } from "./user";

export class Artwork {
  id: number;
  title: string;
  artist: string;
  user: User | null;
  museum: Museum | null;
  creationYear: number;
  description: string;
  imageUrl: string;
  createdAt: string;
  updatedAt: string;

  constructor(
    id: number = 0,
    title: string = '',
    artist: string = '',
    user: User | null = null,
    museums: Museum | null = null,
    creationYear: number = 0,
    description: string = '',
    imageUrl: string = '',
    createdAt: string = '',
    updatedAt: string = ''
  ) {
    this.id = id;
    this.title = title;
    this.artist = artist;
    this.user = user;
    this.museum = museums;
    this.creationYear = creationYear;
    this.description = description;
    this.imageUrl = imageUrl;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
  }

  toString(): string {
    return `Artwork [id=${this.id}, title=${this.title}, artist=${this.artist}, creationYear=${this.creationYear}]`;
  }

  equals(other: Artwork): boolean {
    return this.id === other.id;
  }

  hashCode(): number {
    return this.id;
  }
}
